client/src/components/Connections/dataWrappers/SubgraphWrapper.js

```diff
@@ -13,155 +13,55 @@ import {
-        }
-      })
-    }
-
-    return {
-      // delete x, y to prevent jumping on node load
-      ...props,
-      id,
-      label: bold(poi, `${entity.name} (${entity.related.length})`),
-      group: findGroup(entity),
-      shape: 'infoBox',
-      leaf: false,
-    }
-  })
-  return {nodes, edges, nodeIds}
-}
-
-function transformRaw(rawGraph: {vertices: Array<RawNode>, edges: Array<RawEdge>}): Graph {
-  // transforms graph data for react-graph-vis
-  const {vertices: rawNodes, edges: rawEdges} = rawGraph
-  const nodes: Array<Node> = []
-  const edges: Array<Edge> = []
-  const nodeIds: {[GraphId]: boolean} = {}
-
-  rawNodes.forEach((n: RawNode) => {
-    nodes.push({
-      id: n.eid,
-      label: n.entity_name,
-      is_query: n.query,
-      shape: 'infoBox',
-    })
-    nodeIds[n.eid] = true
-  })
-
-  rawEdges.forEach(([from, to]: RawEdge) => {
-    nodeIds[from] && nodeIds[to] && edges.push({from, to})
-  })
-
-  return {nodes, edges, nodeIds}
+  selectedEids: Array<number>,
+  notable: boolean,
+  preloadNodes: boolean,
 }
 
 const ConnectionWrapper = (WrappedComponent: ComponentType<*>) => {
   const wrapped = (props: SubgraphProps) =>
     isNil(props.subgraph) ? null : <WrappedComponent {...props} />
 
-  return branch(
-    ({entity1, entity2}: EntityProps) => entity1.eids.length > 0,
-    compose(
-      withDataProviders(({entity1, entity2}: EntityProps) => [
-        entity2.query.length > 0
-          ? connectionSubgraphProvider(entity1.eids, entity2.eids, transformRaw)
-          : notableConnectionSubgraphProvider(entity1.eids, transformRaw),
-      ]),
-      // TODO extract selectors
-      connect((state: State, props: EntityProps & ConnectionProps) => ({
-        selectedEids: state.connections.selectedEids,
-        subgraph: enhanceGraph(
-          (props.entity2.query.length > 0
-            ? state.connections.subgraph[
-              `${props.entity1.eids.join()}-${props.entity2.eids.join()}`
-            ]
-            : state.connections.subgraph[`${props.entity1.eids.join()}`]
-          ).data,
-          allEntityDetailsSelector(state),
-          props.connections
-        ),
-        entityDetails: allEntityDetailsSelector(state),
-      })),
-      branch(
-        ({preloadNodes, subgraph}: OwnProps & SubgraphProps) => subgraph != null && preloadNodes,
-        withDataProviders(({subgraph: {nodes}}: SubgraphProps) =>
-          chunk(nodes.map(({id}) => id), MAX_ENTITY_REQUEST_COUNT).map((ids) =>
-            entityDetailProvider(ids, false)
-          )
+  return compose(
+    withDataProviders(({eids1, eids2, notable}: SubgraphProps) => [
+      notable
+        ? notableConnectionSubgraphProvider(eids1, transformRaw)
+        : connectionSubgraphProvider(eids1, eids2 || [], transformRaw),
+    ]),
+    // TODO extract selectors
+    connect((state: State, props: SubgraphProps & ConnectionProps) => ({
+      selectedEids: state.connections.selectedEids,
```

Comment:
I know it's not your code, but in the spirit of graph refactor, you could extract the selectors